lib/vector-core/src/config/mod.rs

```diff
@@ -12,11 +14,23 @@ pub use log_schema::{init_log_schema, log_schema, LogSchema};
 pub const MEMORY_BUFFER_DEFAULT_MAX_EVENTS: usize =
     vector_buffers::config::memory_buffer_default_max_events();
 
-#[derive(Debug, Clone, PartialEq, Copy)]
+// This enum should be kept alphabetically sorted as the bitmask value is used when
+// sorting sources by data type in the GraphQL API.
+#[bitmask(u8)]
 pub enum DataType {
-    Any,
     Log,
     Metric,
+    Trace,
+}
+
+impl fmt::Display for DataType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let mut t = Vec::new();
+        self.contains(DataType::Log).then(|| t.push("Log"));
+        self.contains(DataType::Metric).then(|| t.push("Metric"));
+        self.contains(DataType::Trace).then(|| t.push("Trace"));
+        write!(f, "{}", t.join(","))
```

Comment:
Could this be done without the extra allocations? Does it matter (ie is it performance sensitive)? Note that the final line can be simplified to `f.write_str(t.join(","))`

Reply:
This is not performance sensitive, this was added to get explicit error message when a typing error occur while resolving the topology. I guess it could be rewritten without allocating a vec, but this would probably add a lot of fuss for something out of the hot path.